src/Sarif.Viewer.VisualStudio/Fixes/FixSuggestedActionsSource.cs

```diff
@@ -85,13 +82,22 @@ public void Dispose()
         /// <inheritdoc/>
         public IEnumerable<SuggestedActionSet> GetSuggestedActions(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
         {
-            IEnumerable<SarifErrorListItem> selectedFixableErrors = GetSelectedFixableErrors();
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
Hmm.. Why does this require the UI thread?

Reply:
Not going to hold a PR for this, but we seem to be requiring that we are on the UI thread quite a bit for reasons I don't understand. :)

We should only be switching to UI thread or throwing that we aren't on the UI thread when either 1) VS APIs require it, or 2) we are actively interacting with UI.

In the case of creating objects, requiring the UI thread should never be, well, required.

We should open an issue on this.

---
In reply to: [516036660](https://github.com/microsoft/sarif-visualstudio-extension/pull/239#discussion_r516036660) [](ancestors = 516036660)

Reply:
Good catch. I was able to remove all three calls to `ThrowIfNotOnUIThread` from this class.

---
In reply to: [516038259](https://github.com/microsoft/sarif-visualstudio-extension/pull/239#discussion_r516038259) [](ancestors = 516038259,516036660)

Reply:
I take it back. All three calls are necessary -- removing any of them breaks the build (I don't know what I was seeing a moment ago). I filed [#240](https://github.com/microsoft/sarif-visualstudio-extension/issues/240): "Avoid UI thread requirements on object creation".

---
In reply to: [516105601](https://github.com/microsoft/sarif-visualstudio-extension/pull/239#discussion_r516105601) [](ancestors = 516105601,516038259,516036660)